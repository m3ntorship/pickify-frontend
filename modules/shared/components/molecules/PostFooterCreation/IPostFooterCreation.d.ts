declare namespace IPostFooterCreation {
  export interface IProps {
    handleSelectChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
    togglerIsChecked: boolean;
    disabled: boolean;
    handleTogglerChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  }
}
export { IPostFooterCreation };
