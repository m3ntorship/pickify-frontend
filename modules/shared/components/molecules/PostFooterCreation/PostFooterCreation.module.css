.container {
  @apply md:flex md:justify-between w-full;
}

.primary-button {
  @apply hidden md:flex;
}

.divider {
  @apply md:hidden my-4;
}

.responsive-button {
  @apply md:hidden flex justify-between;
}
