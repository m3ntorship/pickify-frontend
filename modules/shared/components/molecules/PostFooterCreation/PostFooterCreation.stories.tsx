import React from 'react';
import type { Story, Meta } from '@storybook/react';
import PostFooterCreation from './PostFooterCreation';
import type { IPostFooterCreation } from './IPostFooterCreation';

export default {
  title: 'molecules/PostFooterCreation',
  component: PostFooterCreation,
  argTypes: {},
} as Meta;

const Template: Story<IPostFooterCreation.IProps> = (args) => (
  <PostFooterCreation {...args} />
);

export const Default = Template.bind({});
Default.args = {
  togglerIsChecked: false,
  postButtonIsDisabled: true,
};
