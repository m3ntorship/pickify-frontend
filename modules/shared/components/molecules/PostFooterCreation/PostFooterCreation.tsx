import React, { useState } from 'react';
import type { FC, ReactElement } from 'react';
import type { IPostFooterCreation } from './IPostFooterCreation';
import styles from './PostFooterCreation.module.css';
import Privacy from '../Privacy/Privacy';
import Divider from '../../atoms/Divider/Divider';

const PostFooterCreation: FC<IPostFooterCreation.IProps> = ({
  handleSelectChange,
}): ReactElement => {
  const [togglerIsChecked, setTogglerIsChecked] = useState(false);
  const handleTogglerChange = (
    e: React.ChangeEvent<HTMLInputElement>,
  ): void => {
    setTogglerIsChecked(e.target.checked);
  };

  return (
    <div className={styles.container}>
      <Privacy
        privacyOptions={['friends', 'family']}
        withDivider
        togglerIsChecked={togglerIsChecked}
        handleTogglerChange={handleTogglerChange}
        handleSelectChange={handleSelectChange}
      />
      <button type="button" className={styles['primary-button']}>
        post
      </button>
      <div className={styles.divider}>
        <Divider type="horizontal" length="auto" />
      </div>

      <div className={styles['responsive-button']}>
        <button type="button" className={styles['cancel-responsive-button']}>
          cancel
        </button>
        <button type="button" className={styles['primary-responsive-button']}>
          post
        </button>
      </div>
    </div>
  );
};

export default PostFooterCreation;
