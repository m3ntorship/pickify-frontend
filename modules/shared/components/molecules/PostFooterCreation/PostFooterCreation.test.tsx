import * as React from 'react';
import * as renderer from 'react-test-renderer';
import PostFooterCreation from './PostFooterCreation';

describe('Testing Footer with snapshot', () => {
  it('Testing Footer with required props', () => {
    const tree = renderer
      .create(
        <PostFooterCreation
          handlePrivacySelectChange={(): boolean => {
            return true;
          }}
          togglerIsChecked
          handleTheRadioButtonOnChange={(): boolean => {
            return true;
          }}
          postButtonIsDisabled
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Testing Footer with required props', () => {
    const tree = renderer
      .create(
        <PostFooterCreation
          handlePrivacySelectChange={(): boolean => {
            return true;
          }}
          togglerIsChecked
          handleTheRadioButtonOnChange={(): boolean => {
            return true;
          }}
          postButtonIsDisabled={false}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
