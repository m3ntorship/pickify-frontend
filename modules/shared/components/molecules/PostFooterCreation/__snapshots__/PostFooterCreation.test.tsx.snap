// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer with snapshot Testing Footer with required props 1`] = `
<div
  className="container"
>
  <div
    className="privacyContainer"
  >
    <div
      className="hideMyIdentityContainer"
    >
      <label
        className="toggleBodyClasses toggleBodyClassesSmall toggleBodyClassesCheckedAndEnabled toggleBodyClassesSmallChecked"
        htmlFor="hideMyIdentityToggler"
      >
        <input
          className="opacity-0 absolute checkBoxInputEnabled"
          data-testid="hideMyIdentityToggler"
          defaultChecked={true}
          disabled={false}
          id="hideMyIdentityToggler"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          className="circleClasses circleClassesChecked circleClassesCheckedSmall circleClassesSmall"
        />
      </label>
      <span
        className="hideMyIdentityText"
      >
        Hide my identity
      </span>
    </div>
    <div
      className="divider"
      data-testid="dividerTestId"
    >
      <div
        className="bg-grey-shd6 w-px"
        style={
          Object {
            "height": "16px",
          }
        }
      />
    </div>
    <div
      className="privacySelectorContainer"
    >
      <span
        className="privacySelectorText"
      >
        Privacy:
      </span>
      <select
        className="privacySelector"
        data-testid="selectTestId"
        onChange={[Function]}
      >
        <option
          className="privacySelectorOptions"
          data-testid="friends"
          value="friends"
        >
          friends
        </option>
        <option
          className="privacySelectorOptions"
          data-testid="family"
          value="family"
        >
          family
        </option>
      </select>
    </div>
  </div>
  <div
    className="primary-button"
  >
    <button
      className="btn btn-primary btn-small"
      disabled={true}
      type="submit"
    >
      post
    </button>
  </div>
  <div
    className="divider"
  >
    <div
      className="bg-grey-shd6 h-px"
      style={
        Object {
          "width": "auto",
        }
      }
    />
  </div>
  <div
    className="responsive-button"
  >
    <button
      className="btn btn-normal"
      disabled={false}
      type="button"
    >
      cancel
    </button>
    <button
      className="btn btn-primary btn-xlarge"
      disabled={true}
      type="submit"
    >
      post
    </button>
  </div>
</div>
`;

exports[`Testing Footer with snapshot Testing Footer with required props 2`] = `
<div
  className="container"
>
  <div
    className="privacyContainer"
  >
    <div
      className="hideMyIdentityContainer"
    >
      <label
        className="toggleBodyClasses toggleBodyClassesSmall toggleBodyClassesCheckedAndEnabled toggleBodyClassesSmallChecked"
        htmlFor="hideMyIdentityToggler"
      >
        <input
          className="opacity-0 absolute checkBoxInputEnabled"
          data-testid="hideMyIdentityToggler"
          defaultChecked={true}
          disabled={false}
          id="hideMyIdentityToggler"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          className="circleClasses circleClassesChecked circleClassesCheckedSmall circleClassesSmall"
        />
      </label>
      <span
        className="hideMyIdentityText"
      >
        Hide my identity
      </span>
    </div>
    <div
      className="divider"
      data-testid="dividerTestId"
    >
      <div
        className="bg-grey-shd6 w-px"
        style={
          Object {
            "height": "16px",
          }
        }
      />
    </div>
    <div
      className="privacySelectorContainer"
    >
      <span
        className="privacySelectorText"
      >
        Privacy:
      </span>
      <select
        className="privacySelector"
        data-testid="selectTestId"
        onChange={[Function]}
      >
        <option
          className="privacySelectorOptions"
          data-testid="friends"
          value="friends"
        >
          friends
        </option>
        <option
          className="privacySelectorOptions"
          data-testid="family"
          value="family"
        >
          family
        </option>
      </select>
    </div>
  </div>
  <div
    className="primary-button"
  >
    <button
      className="btn btn-primary btn-small"
      disabled={false}
      type="submit"
    >
      post
    </button>
  </div>
  <div
    className="divider"
  >
    <div
      className="bg-grey-shd6 h-px"
      style={
        Object {
          "width": "auto",
        }
      }
    />
  </div>
  <div
    className="responsive-button"
  >
    <button
      className="btn btn-normal"
      disabled={false}
      type="button"
    >
      cancel
    </button>
    <button
      className="btn btn-primary btn-xlarge"
      disabled={false}
      type="submit"
    >
      post
    </button>
  </div>
</div>
`;
